packages/finch-graphql/src/client/FinchClient.ts

```diff
@@ -0,0 +1,54 @@
+import { DocumentNode } from 'graphql';
+import gql from 'graphql-tag';
+import { QueryCache, Listener } from './cache';
+import { FinchQueryOptions, GenericVariables } from '../types';
+import { isDocumentNode } from '../utils';
+import { queryApi } from './client';
+
+interface FinchClientOptions {
+  cache?: QueryCache;
+  id?: string;
+  messageKey?: string;
+}
+
+export class FinchClient {
+  private cache: QueryCache | undefined;
+  private id: string | undefined;
+  private messageKey: string | undefined;
+
+  constructor(options: FinchClientOptions) {
+    this.cache = options.cache;
+    this.id = options.id;
+    this.messageKey = options.messageKey;
+  }
+
+  async query<Query extends {} = {}, Variables extends GenericVariables = {}>(
```

Comment:
Should we make a mutation one too? It's the same thing but might help on terms of caching mechanisms.